xandikos/web.py

```diff
@@ -1030,6 +1030,14 @@ def get_resource(self, relpath):
             except KeyError:
                 return None
 
+    def set_principal(self, user):
+        principal = "/%s/" % user
+        if not self.get_resource(principal):
+            if os.getenv("AUTOCREATE"):
```

Comment:
This class should ideally not be looking at the environment. Perhaps pass in a variable to the class constructor?

Reply:
Will do.